src/App.tsx

```diff
@@ -1,26 +1,183 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { Filters } from './types/Filters';
+import { ErrorTypes } from './types/ErrorTypes';
+import { filterTodos } from './utils/filterTodos';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoError } from './components/TodoError';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorTodo, setErrorTodo] = useState<ErrorTypes>(ErrorTypes.Empty);
+  const [currentFilter, setCurrentFilter] = useState<Filters>(Filters.All);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [hasTitleFocus, setHasTitleFocus] = useState(false);
+
+  const titleRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    titleRef.current?.focus();
+  }, [hasTitleFocus]);
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorTodo(ErrorTypes.LoadTodo));
+  }, []);
+
+  useEffect(() => {
+    if (errorTodo === ErrorTypes.Empty) {
+      return;
+    }
+
+    const timeoutId = setTimeout(() => {
+      setErrorTodo(ErrorTypes.Empty);
+    }, 3000);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [errorTodo]);
+
+  const filteredTodos = useMemo(
+    () => filterTodos(todos, currentFilter),
+    [todos, currentFilter],
+  );
+
+  function onAdd({ userId, title, completed }: Omit<Todo, 'id'>) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const onAdd = ({ userId, title, completed }: Omit<Todo, 'id'>) => {
```